src/huntsman/pocs/autoflats.py

```diff
@@ -0,0 +1,448 @@
+            exptime = self._max_exptime
+        if exptime <= self._min_exptime:
+            self.logger.warning(f"Required exptime for {camera_name} is less than the allowable"
+                                " minimum.")
+            exptime = self._min_exptime
+
+        return exptime
+
+    def _get_average_counts(self, filename_dict):
+        """ Calculate the average counts for each camera.
+        Args:
+            filename_dict (dict): dict of cam_name: filename pairs.
+        Returns:
+            dict: cam_name: average_counts pairs.
+        """
+        counts = {}
+        for cam_name, filename in filename_dict.items():
+            counts[cam_name] = self._get_average_count(cam_name, filename)
+            self.logger.debug(f"Average counts for {cam_name}: {counts[cam_name]:.1f}")
+        return counts
+
+    def _get_average_count(self, cam_name, filename, min_counts=1):
+        """ Read the data and calculate a clipped-mean count rate.
+        Args:
+            filename (str): The filename containing the data.
+            bias (float): The bias level to subtract from the image.
+            min_counts (float): The minimum count rate returned by this function. Can cause
+                problems if less than or equal to 0, so 1 (default) is a safe choice.
+        Returns:
+            float: The average counts.
+        """
+        try:
+            data = self._load_fits_data(filename)
+        except FileNotFoundError:
+            data = self._load_fits_data(filename + ".fz")
+
+        # Calculate average counts per pixel
+        average_counts, _, _ = sigma_clipped_stats(data - self._biases[cam_name])
+        if average_counts < min_counts:
+            self.logger.warning('Clipping mean flat-field counts at minimum value: '
+                                f'{average_counts}<{min_counts}.')
+            average_counts = min_counts
+
+        return average_counts
+
+    def _take_biases(self, seconds=0):
+        """ Take a bias exposure for each camera. Should only be run once per sequence.
+        Args:
+            seconds (optional, float): The exposure time in seconds, default 0.
+            dtype (str): The data type for the exposure data, default float32.
+        """
+        self.logger.info("Taking biases for auto-flat fielding.")
+        futures = []
+        filenames = {}
+
+        # Take the bias exposures
+        for cam_name, camera in self.cameras.items():
+            filename = self._get_bias_filename(camera)
+            future = camera.take_exposure(filename=filename, seconds=seconds, dark=True)
+            futures.append(future)
+            filenames[cam_name] = filename
+
+        # Wait for exposures to complete
+        wait_for_futures(futures)
+
+        # Read the biases
+        self._biases = {}
+        for cam_name, filename in filenames.items():
+            self._biases[cam_name] = self._load_fits_data(filename)
+
+    def _take_darks(self, **kwargs):
```

Comment:
Why having another function to take darks when there's already one (i.e. `take_dark_images` in `huntsman.pocs.observatory`)? Is it not better just importing that function here in `autoflats.py` and use it as it is intended to be used? Otherwise that function would only be used within the `taking_darks` state and this function `_take_darks` would just be duplicated code.